lib/fluent/plugin/in_kafka_group.rb

```diff
@@ -254,6 +255,9 @@ def reconnect_consumer
     log.error "unexpected error during re-starting consumer object access", :error => e.to_s
     log.error_backtrace
     if @retry_count <= @retry_limit or disable_retry_limit
+      # Adding the following line to ensure, consumer object is cleaned before recursively calling reconnect_consumer.
+      # This ensures no memory leak.
+      consumer = nil
       reconnect_consumer
```

Comment:
Probably just calling `retry` here is better than resetting `consumer` and calling `reconnect_consumer` recursively.